api/konnect/v1alpha1/konnect_gateway_controlplane_types.go

```diff
@@ -80,11 +80,16 @@ type KonnectGatewayControlPlaneSpec struct {
 
 // MirrorSpec contains the Konnect Mirror configuration.
 type MirrorSpec struct {
+	Konnect MirrorKonnect `json:"konnect"`
```

Comment:
Let's add a required code marker here if the field is required and add a godoc for this field.

Reply:
The `required` code marker is not needed as the struct is not a pointer.
Adding godoc

Reply:
Recently we've been trying to be consistent about this (perhaps somewhat implicitly), trying to following https://github.com/kubernetes-sigs/kube-api-linter/?tab=readme-ov-file#optionalorrequired.

I'm OK leaving this as is until rules in the linter solidify and its usage becomes more mainstream.

Reply:
oh, good to know 👍🏻 I've added it in 54f01f5bc53b4919f8017fda41d63d36cba33204